.editorconfig

```diff
@@ -0,0 +1,35 @@
+# EditorConfig is awesome: http://EditorConfig.org
+
+# top-most EditorConfig file
+root = true
+
+# Unix-style newlines with a newline ending every file
+[*]
+end_of_line = lf
+insert_final_newline = true
+charset = utf-8
+indent_style = space
+indent_size = 2
+trim_trailing_whitespace = false
+
+# Keep Makefiles sane
+[Makefile*]
+indent_style = tab
+indent_size = 4
+
+# Ensure specific settings for YAML (and similar) in case defaults change
+[{*.yml,*.sublime-syntax}]
+indent_style = space
+indent_size = 2
+
+[{*.tmPreferences,*.sublime-snippet}]
+indent_style = tab
+indent_size = 2
```

Comment:
I think having a tab rendered with width 4 is most common nowadays (and also the default for Sublime Text), so I believe this shouldn't be changed. It should be using tabs, but there is no need to specify the indent size.

Reply:
`indent_size` can be removed across the board then, except for `{*.yml,*.sublime-syntax}`. The Makefile section should perhaps also not be there, since there is only one Makefile in the repo apart from the whole Makefile package.

If we go for the minimum, I guess it will be:
```
root = true

[{*.yml,*.sublime-syntax}]
indent_style = space
indent_size = 2

[{*.tmPreferences,*.sublime-snippet}]
indent_style = tab

# Syntax test files should use spaces to work correctly.
[syntax_test_*]
trim_trailing_whitespace = false
indent_style = space
```

Reply:
This looks better :+1:

We might even recommend able to add the newline at eof. I'd like to check if any file doesn't do that currently though.

Reply:
s/recommend/be/

Reply:
Seems there are 7 files that do not end with a new line:
```
% git ls-files | while read f; do tail -c1 "$f" | wc -l | grep -q "1" || echo "$f"; done                                                                     ActionScript/ActionScript.sublime-build
JavaScript/JavaScript.sublime-syntax
Lua/Lua.sublime-build
Ruby/Default.sublime-keymap
Scala/info.plist
Scala/syntax_test_scala.scala
Text/Snippets/lorem.sublime-snippet
```

Reply:
>`Scala/info.plist`

Wow, this does nothing and can be removed.

To all other files a newline could easily be added at the end. This would be a safe change, but it should be made together with the change to standardize the editorconfig configuration.